### PR TITLE
[breaking] Refactor the BLE devices identification

### PR DESCRIPTION
**Pull request by 1technophile**
generalize the use of the mac address with : for the lists, white, black and internal identification
Move structures from config_BT.h to ZgatewayBT.ino
update doc

The goal is to align the vector array ids format with the one extracted from BLEscan

### CODE REVIEW

**Comment by jmw6773** [comment]
If the `PublishDeviceData`, `createOrUpdateDevice`, and `launch_discovery` functions all accepted a BLEdevice object as a parameter, the current BLEdevice can passed among them to save repetitive calls to the `getDeviceByMac`.

The `updateWorB` function would still need to do a `getDeviceByMac` lookup for each device in the White/Black list JSON before calling the `createOrUpdateDevice` function though.

**Comment by 1technophile** [comment]
Good catch!
I see the point for `createOrUpdateDevice` and `launch_discovery` but regarding `PublishDeviceData` how do you think it could be optimized?

**Comment by jmw6773** [comment]
The `PublishDeviceData` function itself wouldn't be optimized, but as it calls the `launch_discovery` function, it would need the BLEdevice passed into it so it could pass it along to the 'launch_discovery' function.

**Comment by jmw6773** [comment]
Minor optimization as `abs` isn't intensive, but `abs(Minrssi)` could be moved to the `setupBT()` and `MQTTtoBT` functions. (Where the value is set instead of every time it is used.)

Remove from here:
https://github.com/1technophile/OpenMQTTGateway/blob/7715bad2f4da9bc54ef4519b34237e97b44d2efb/main/ZgatewayBT.ino#L690


Add abs() in these 3 locations:
https://github.com/1technophile/OpenMQTTGateway/blob/4f84e08219498102471e17493e728e4a6fdbb5c8/main/ZgatewayBT.ino#L536

https://github.com/1technophile/OpenMQTTGateway/blob/4f84e08219498102471e17493e728e4a6fdbb5c8/main/ZgatewayBT.ino#L1036

The ESP32 code relies on the `Minrssi` variable being set in the header file. Should be moved to INO file for consistancy.
https://github.com/1technophile/OpenMQTTGateway/blob/7715bad2f4da9bc54ef4519b34237e97b44d2efb/main/ZgatewayBT.ino#L505

File: main/ZgatewayBT.ino [hunk 1]
```diff
@@ -590,23 +594,20 @@ bool BTtoMQTT() {
...
             JsonObject& BLEdata = jsonBuffer.createObject();
             strupp(d[0].extract);
-
-            BLEdevice* device = getDeviceByMac(d[0].extract);
+            String Id;
+            for (int i = 0; i < 12; i++) {
+              Id += String(d[0].extract[i]);
+              if (((i - 1) % 2 == 0) && (i != 11))
+                Id += ":";
+            }
```
[the remark this answers is not in the document]
Comment (1technophile) [reply main/ZgatewayBT.ino]: done thanks

**Comment by 1technophile** [comment]
> Minor optimization as abs isn't intensive, but abs(Minrssi) could be moved to the setupBT() and MQTTtoBT functions. (Where the value is set instead of every time it is used.)

Done

**Comment by 1technophile** [comment]
> The PublishDeviceData function itself wouldn't be optimized, but as it calls the launch_discovery function, it would need the BLEdevice passed into it so it could pass it along to the 'launch_discovery' function.

At final I didn't implemented this change on this PR as I saw also impact on the `BLEdevice` creation into `createOrUpdateDevice` due to the removal of `mac `and it looked like the begining of a little ball of yarn. Thanks for the proposal I keep it in mind.

**Comment by 1technophile** [comment]
It should have been a rebase... let,s try next time

**Pull request merged**